src/textual/widgets/_directory_tree.py

```diff
@@ -375,7 +375,7 @@ def _populate_node(self, node: TreeNode[DirEntry], content: Iterable[Path]) -> N
             )
         node.expand()
 
-    def _directory_content(self, location: Path, worker: Worker) -> Iterator[Path]:
+    def _directory_content(self, location: Path, worker: Worker[Any]) -> Iterator[Path]:
```

Comment:
If I'm reading the code correctly, this `worker` instance comes from `_load_directory`, so it would be a `Worker[list[Path]]`, which is a bit more specific than `Worker[Any]`.

Reply:
`_directory_content` is called with the result of `get_current_worker()` which returns (and can only return) a `Worker[Any]`, so this would cause an error in `_load_directory`

Reply:
Of course I might be misunderstanding something, but what I'm suggesting is that we narrow the type ourselves, which should work.

E.g., [the code below typechecks](https://mypy-play.net/?mypy=latest&python=3.12&gist=3f1ed4ae05b311ea040a72651cea8fb5):

```py
from typing import Any

def any_factory() -> Any:
    ...

integer: int = any_factory()
```

`get_current_worker` is being called inside a worker, so we know that the current worker is the function we're in, so we know its type, right?
That was my reasoning.

Reply:
Sorry for not getting back to you earlier, it's been a hectic week! But yes, you're right. I have a bad habit of thinking of `Any` as the base type, in which case this would have not worked due to invariance, but it's actually an opt-out of the typing system. Personally I always avoid implicit casts from `Any`, since it means the cast goes completely unchecked and it makes it easy to introduce errors while still being under a false sense of security.

Incidentally, this reminded me of something that I'd had in the back of my mind for a while as something I've been wanting to try in practice, namely this note on `Any` vs `object` from [this page on typing best practices](https://typing.readthedocs.io/en/latest/source/best_practices.html#ergonomic-practices):

> Generally, use Any when a type cannot be expressed appropriately with the current type system or using the correct type is unergonomic.
>
> If a function accepts every possible object as an argument, for example because it’s only passed to str(), use object instead of Any as type annotation. Similarly, if the return value of a callback is ignored, annotate it with object

I actually did have a go this week to see if I could eliminate `Any` in favour of `object` in my own codebase for better typing safety, and while I wasn't able to remove all uses due to limitations in Python's typing system (notably the [lack of wildcards](https://github.com/python/typing/issues/912)), I did have decent success and I'm definitely more confident in the type checking now.

If that's something you would be interested in I'd be happy to update this PR to use `Worker[object]` instead of `Worker[Any]` where applicable, which would provide better type safety for users of the code and make clear where explicit casts need to be made. It would mean users having to add an explicit cast if they want to get the result of `get_current_worker()` as a `Worker[int]` or whatever, but I don't think that's an issue - what would you even use the result type for from the worker itself before the result has been returned? `is_cancelled` can be checked even if you don't have the specific type.

Anyway! To get back to the point of your comment, there are three options I see here, in descending order of personal preference:

* Keep the argument as `Worker[Any]` (or better, `Worker[object]`)
    * We're only using the worker to access `is_cancelled`, and I don't think there's any real reason to add more specialisation than we actually need for the function - less casting means more type safety
* Have `get_current_worker` return `Worker[object]` but let `_directory_content` take a `Worker[Path]`, with an explicit cast to silence the type checker
    * Would at least make it clear that there is casting going on, and it makes it easy to do a search for `cast` later if one wanted to eliminate casting for the sake of stronger typing guarantees in the future
* Stick to `Worker[Any]` for `get_current_worker` and rely on implicit casting to `Worker[Path]`
    * Not a fan personally! Hard to spot and easy for issues to slip in. Only real way of tracking this down in the future is to run Mypy with `--disallow-any-expr` which is impossibly noisy

Let me know your thoughts!

Reply:
Thanks for your comments!
I wasn't aware of the best practice of using `object` (instead of `Any`) in the contexts listed but it does make sense.
I _think_ we'd better leave it as `Worker[Any]` to be in line with the remainder of the code base.

I'll muster the courage to review this PR and then I'll rope someone else in to also take a look!